### ISSUE TITLE
[BUG] Supported minimum K8s version not correct

### ISSUE THREAD

**Issue by chriscchien** [longhorn/longhorn#7196]
## Describe the bug (🐛 if you encounter this issue)

From official [website](https://longhorn.io/docs/1.5.3/deploy/important-notes/#supported-kubernetes-versions) and chart value ([master](https://github.com/longhorn/longhorn/blob/57bf3edb1faa955ab077187a518d7be99ff38792/chart/Chart.yaml#L5), [v1.5.3](https://github.com/longhorn/longhorn/blob/90b9d194f3ea23e0f3f7a156f8c9a09b599adeb2/chart/Chart.yaml#L5)), Longhorn can install on K8s v1.21, but not.

## To Reproduce

1. `kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/longhorn.yaml` on `v1.21.5+k3s1` cluster
2. Pod `longhorn-driver-deployer` stuck at init

## Expected behavior

Update supported K8s version

## Environment

 - Longhorn version: v1.6.0-dev
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): kubectl and helm
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: v1.21.5+k3s1

## Additional context

N/A

**Comment by innobead** [comment]
@c3y1huang Please quickly check this one, even though we will update the min K8s version later.

**Comment by innobead** [comment]
@chriscchien the original severity label is incorrect because it should be `severity/1` instead. 

cc @longhorn/qa

**Comment by c3y1huang** [comment]
Test result:
- v1.5.3:
  ```shell
  > k get nodes
  NAME           STATUS   ROLES                  AGE     VERSION
  ip-10-0-1-46   Ready    control-plane,master   2m49s   v1.21.5+k3s1
  
  > k -n longhorn-system get pod
  NAME                                                READY   STATUS    RESTARTS   AGE
  longhorn-ui-5694cc6c58-gkvwn                        1/1     Running   0          2m3s
  longhorn-ui-5694cc6c58-knm4w                        1/1     Running   0          2m3s
  longhorn-manager-lzsvb                              1/1     Running   0          2m3s
  longhorn-driver-deployer-7767786ddb-vzwml           1/1     Running   0          2m3s
  csi-provisioner-579866cdd8-vjvcr                    1/1     Running   0          92s
  csi-attacher-868487bdf9-f2svk                       1/1     Running   0          92s
  csi-provisioner-579866cdd8-fjlmv                    1/1     Running   0          92s
  csi-snapshotter-9b68bbfb8-dl4ft                     1/1     Running   0          92s
  csi-attacher-868487bdf9-7pdkc                       1/1     Running   0          92s
  csi-snapshotter-9b68bbfb8-6rmn9                     1/1     Running   0          92s
  csi-resizer-cdd748db8-jls9v                         1/1     Running   0          92s
  csi-resizer-cdd748db8-clh5l                         1/1     Running   0          92s
  csi-attacher-868487bdf9-vn8sf                       1/1     Running   0          92s
  csi-provisioner-579866cdd8-n556z                    1/1     Running   0          92s
  csi-snapshotter-9b68bbfb8-z57qx                     1/1     Running   0          92s
  instance-manager-002ed4bc07e37c5861d3a0baa397106e   1/1     Running   0          98s
  csi-resizer-cdd748db8-g99mg                         1/1     Running   0          92s
  engine-image-ei-68f17757-9b9lx                      1/1     Running   0          98s
  longhorn-csi-plugin-4mk9p                           3/3     Running   0          91s
  ```
 - master ([866f1155f74037bb41bbaf3d4f8e3707f71b573f](https://github.com/longhorn/longhorn/tree/866f1155f74037bb41bbaf3d4f8e3707f71b573f))
   ```shell
   > k get nodes
   NAME           STATUS   ROLES                  AGE   VERSION
   ip-10-0-1-46   Ready    control-plane,master   13m   v1.21.5+k3s1
   
   > k -n longhorn-system get pod
   NAME                                                READY   STATUS    RESTARTS   AGE
   longhorn-ui-b8766f754-hzxhj                         1/1     Running   0          100s
   longhorn-ui-b8766f754-n5jzs                         1/1     Running   0          100s
   longhorn-manager-7qpht                              1/1     Running   0          100s
   longhorn-driver-deployer-74f79c586d-4tk7l           1/1     Running   0          100s
   csi-resizer-858b9cb466-xzp5k                        1/1     Running   0          65s
   csi-snapshotter-6756d6d84-qgrrz                     1/1     Running   0          64s
   csi-provisioner-5bfb9d8577-zbtzp                    1/1     Running   0          65s
   csi-snapshotter-6756d6d84-smg6c                     1/1     Running   0          64s
   csi-attacher-66d89c889-wwn6m                        1/1     Running   0          65s
   instance-manager-ecd9e6c7760c50378c9f09dcad435f7b   1/1     Running   0          71s
   csi-provisioner-5bfb9d8577-7thqn                    1/1     Running   0          65s
   csi-resizer-858b9cb466-cv7zn                        1/1     Running   0          65s
   csi-attacher-66d89c889-wdgbd                        1/1     Running   0          65s
   csi-snapshotter-6756d6d84-xrqxs                     1/1     Running   0          64s
   csi-resizer-858b9cb466-q5w5w                        1/1     Running   0          65s
   csi-provisioner-5bfb9d8577-t5f24                    1/1     Running   0          65s
   csi-attacher-66d89c889-zcm5l                        1/1     Running   0          65s
   engine-image-ei-b907910b-4n5vw                      1/1     Running   0          71s
   longhorn-csi-plugin-rlrst                           3/3     Running   0          64s
   ```

**Comment by chriscchien** [comment]
Update, I can reproduce this symptom on `Ubuntu 22.04` with `v1.21.5+k3s1`. But can not reproduce on `SLES15 sp5` with `v1.21.5+k3s1`.
 Longhorn master can success deploy on `SLES15 sp5` with `v1.21.5+k3s1`

**Comment by c3y1huang** [comment]
> @chriscchien have we seen what reason about `longhorn-driver-deployer` failed to run? `kubectl describe` or any log from the init container?

On @chriscchien's environment, the driver-deployer cannot start due to the initContainer being unable to successfully [curl to the backend](https://github.com/longhorn/longhorn/blob/master/chart/templates/deployment-driver.yaml#L20). It seems that the nodes cannot curl between nodes; while curling from the backend node yielded a 200 response, and attempts from other nodes returned 000 response.

**Comment by chriscchien** [comment]
> @chriscchien how did you set up that cluster? It seems like an env issue, and the cluster CNI network seems not working properly. You should just try our pipeline, which I remember we can choose what distro version you want to run.
> 
> cc @yangchiu

Here is pipeline with k3s distro = `v1.21.5+k3s1` run result
- SLES15
   - build [5338](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/5338/) - Longhorn install success, longhorn-test running
- Ubuntu 22.04
  - build [5336](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/5336/) - Longhorn install timeout
  - build [5337](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/5337/) - Longhorn install timeout

- My environment setup steps:

<details>



 1. Create fresh VM (3 * Ubuntu 22.04/SLES 15)
 2. Choose one node as master
```
curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v1.21.5+k3s1 sh -s - --write-kubeconfig-mode 644  
sudo cat /var/lib/rancher/k3s/server/node-token
```
 3. On the other nodes
```
curl -sfL https://get.k3s.io | K3S_URL=https://<master node IP>:6443 K3S_TOKEN=<token from step2> INSTALL_K3S_VERSION=v1.21.5+k3s1 sh -
```
 4. Install iSCSI and nfs-common
```
kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/prerequisite/longhorn-iscsi-installation.yaml
kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/prerequisite/longhorn-nfs-installation.yaml
```
 5. Install Longhorn

</details>

**Comment by c3y1huang** [comment]
I can reproduce the DNS resolution issue using v1.21.5+k3s1 on Ubuntu 22.04, and manage to resolve it by including `--flannel-backend=ipsec` during k3s installation.
```bash
curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v1.21.5+k3s1 sh -s - --write-kubeconfig-mode 644 --flannel-backend=ipsec
```
Related upstream discussion: https://github.com/k3s-io/k3s/issues/6171

```
> cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy

> kubectl -n longhorn-system get pod
NAME                                                READY   STATUS              RESTARTS   AGE
longhorn-ui-b8766f754-nzcq7                         1/1     Running             0          11m
longhorn-ui-b8766f754-q8n8m                         1/1     Running             0          11m
longhorn-manager-tjhf5                              1/1     Running             2          11m
longhorn-driver-deployer-74f79c586d-d5l6k           1/1     Running             0          11m
longhorn-manager-cw8hm                              1/1     Running             2          11m
engine-image-ei-b907910b-7k8pr                      1/1     Running             0          10m
engine-image-ei-b907910b-m6h8q                      1/1     Running             0          10m
csi-provisioner-5bfb9d8577-jprk4                    1/1     Running             0          10m
csi-attacher-66d89c889-jvxct                        1/1     Running             0          10m
csi-snapshotter-6756d6d84-lqq78                     1/1     Running             0          10m
csi-resizer-858b9cb466-9qtrf                        1/1     Running             0          10m
csi-resizer-858b9cb466-9w25m                        1/1     Running             0          10m
csi-attacher-66d89c889-5nhwf                        1/1     Running             0          10m
csi-provisioner-5bfb9d8577-z54fq                    1/1     Running             0          10m
csi-snapshotter-6756d6d84-497h8                     1/1     Running             0          10m
instance-manager-eacc98ef42202655177ffaf0a37039e4   1/1     Running             0          10m
longhorn-manager-9kcqh                              1/1     Running             2          11m
csi-resizer-858b9cb466-vj2nv                        1/1     Running             0          10m
csi-attacher-66d89c889-ncd9w                        1/1     Running             0          10m
csi-provisioner-5bfb9d8577-rhksd                    1/1     Running             0          10m
csi-snapshotter-6756d6d84-lkprk                     1/1     Running             0          10m
engine-image-ei-b907910b-nm5zj                      1/1     Running             0          10m
longhorn-csi-plugin-nkdrg                           3/3     Running             0          10m
longhorn-csi-plugin-784xb                           3/3     Running             0          10m
longhorn-csi-plugin-9tdgq                           3/3     Running             0          10m
instance-manager-64e9ba4eb1994485be433fd978e8dc90   1/1     Running             0          9m55s
instance-manager-e6f0c37f458335a1b3a510bfe6c24cac   1/1     Running             0          9m39s
```

**Comment by innobead** [comment]
cc @longhorn/qa

**Comment by c3y1huang** [comment]
Here is the related flannel fix: https://github.com/flannel-io/flannel/pull/1485

K3s should have this fixed backported to [v1.21.6+k3s1](https://github.com/k3s-io/k3s/issues/4188#issuecomment-963124378).

@chriscchien could you give the following steps a try?
1. Run `apt upgrade` on all cluster nodes.
1. Reboot all cluster nodes.
1. install k3s v1.21.6+k3s1.
   ```bash
   curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v1.21.6+k3s1 sh -s - --write-kubeconfig-mode 644  
   sudo cat /var/lib/rancher/k3s/server/node-token
   ```
1. Install iSCSI and NFS-common.
   ```bash
   kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/prerequisite/longhorn-iscsi-installation.yaml
   kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/prerequisite/longhorn-nfs-installation.yaml
   ```
1. Install longhorn.

Expected result:
```
> cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy

> k -n longhorn-system get pod
NAME                                                READY   STATUS    RESTARTS   AGE
longhorn-ui-b8766f754-h7fsf                         1/1     Running   0          105s
longhorn-ui-b8766f754-2vgqt                         1/1     Running   0          105s
longhorn-manager-zs9ss                              1/1     Running   0          105s
longhorn-manager-hprx6                              1/1     Running   0          105s
longhorn-manager-7t45g                              1/1     Running   1          105s
longhorn-driver-deployer-74f79c586d-mhz5x           1/1     Running   0          105s
instance-manager-e6f0c37f458335a1b3a510bfe6c24cac   1/1     Running   0          84s
instance-manager-64e9ba4eb1994485be433fd978e8dc90   1/1     Running   0          82s
instance-manager-eacc98ef42202655177ffaf0a37039e4   1/1     Running   0          83s
engine-image-ei-b907910b-v6hjw                      1/1     Running   0          84s
engine-image-ei-b907910b-nmq8f                      1/1     Running   0          84s
engine-image-ei-b907910b-7wckr                      1/1     Running   0          84s
csi-resizer-858b9cb466-gtrrf                        1/1     Running   0          72s
csi-provisioner-5bfb9d8577-m7kqs                    1/1     Running   0          72s
csi-attacher-66d89c889-wr9gg                        1/1     Running   0          72s
csi-snapshotter-6756d6d84-c8d2x                     1/1     Running   0          71s
csi-snapshotter-6756d6d84-r87xv                     1/1     Running   0          71s
csi-attacher-66d89c889-5rpdc                        1/1     Running   0          72s
csi-resizer-858b9cb466-xljzt                        1/1     Running   0          72s
csi-provisioner-5bfb9d8577-gzcrj                    1/1     Running   0          72s
csi-resizer-858b9cb466-grrvp                        1/1     Running   0          72s
csi-snapshotter-6756d6d84-mhk9g                     1/1     Running   0          71s
csi-provisioner-5bfb9d8577-bq4v8                    1/1     Running   0          72s
csi-attacher-66d89c889-wmnnl                        1/1     Running   0          72s
longhorn-csi-plugin-bwhts                           3/3     Running   0          71s
longhorn-csi-plugin-lh88t                           3/3     Running   0          71s
longhorn-csi-plugin-ntz9f                           3/3     Running   0          71s
```

**Comment by chriscchien** [comment]
> Here is the related flannel fix: [flannel-io/flannel#1485](https://github.com/flannel-io/flannel/pull/1485) and K3s should have this fix backported to [v1.21.6+k3s1](https://github.com/k3s-io/k3s/issues/4188#issuecomment-963124378).
> 
> @chriscchien could you give the following steps a try on Ubuntu 22.04?
> 
> 1. Run `apt upgrade` on all cluster nodes.
> 2. Reboot all cluster nodes.
> 3. install k3s version >= v1.21.6+k3s1.
>    ```shell
>    curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v1.21.6+k3s1 sh -s - --write-kubeconfig-mode 644  
>    sudo cat /var/lib/rancher/k3s/server/node-token
>    ```
> 4. Install iSCSI and NFS-common.
>    ```shell
>    kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/prerequisite/longhorn-iscsi-installation.yaml
>    kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/prerequisite/longhorn-nfs-installation.yaml
>    ```
> 5. Install longhorn.
> 
> Expected result:
> 
> ```
> > cat /etc/os-release 
> PRETTY_NAME="Ubuntu 22.04.3 LTS"
> NAME="Ubuntu"
> VERSION_ID="22.04"
> VERSION="22.04.3 LTS (Jammy Jellyfish)"
> VERSION_CODENAME=jammy
> ID=ubuntu
> ID_LIKE=debian
> HOME_URL="https://www.ubuntu.com/"
> SUPPORT_URL="https://help.ubuntu.com/"
> BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
> PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
> UBUNTU_CODENAME=jammy
> 
> > k -n longhorn-system get pod
> NAME                                                READY   STATUS    RESTARTS   AGE
> longhorn-ui-b8766f754-h7fsf                         1/1     Running   0          105s
> longhorn-ui-b8766f754-2vgqt                         1/1     Running   0          105s
> longhorn-manager-zs9ss                              1/1     Running   0          105s
> longhorn-manager-hprx6                              1/1     Running   0          105s
> longhorn-manager-7t45g                              1/1     Running   1          105s
> longhorn-driver-deployer-74f79c586d-mhz5x           1/1     Running   0          105s
> instance-manager-e6f0c37f458335a1b3a510bfe6c24cac   1/1     Running   0          84s
> instance-manager-64e9ba4eb1994485be433fd978e8dc90   1/1     Running   0          82s
> instance-manager-eacc98ef42202655177ffaf0a37039e4   1/1     Running   0          83s
> engine-image-ei-b907910b-v6hjw                      1/1     Running   0          84s
> engine-image-ei-b907910b-nmq8f                      1/1     Running   0          84s
> engine-image-ei-b907910b-7wckr                      1/1     Running   0          84s
> csi-resizer-858b9cb466-gtrrf                        1/1     Running   0          72s
> csi-provisioner-5bfb9d8577-m7kqs                    1/1     Running   0          72s
> csi-attacher-66d89c889-wr9gg                        1/1     Running   0          72s
> csi-snapshotter-6756d6d84-c8d2x                     1/1     Running   0          71s
> csi-snapshotter-6756d6d84-r87xv                     1/1     Running   0          71s
> csi-attacher-66d89c889-5rpdc                        1/1     Running   0          72s
> csi-resizer-858b9cb466-xljzt                        1/1     Running   0          72s
> csi-provisioner-5bfb9d8577-gzcrj                    1/1     Running   0          72s
> csi-resizer-858b9cb466-grrvp                        1/1     Running   0          72s
> csi-snapshotter-6756d6d84-mhk9g                     1/1     Running   0          71s
> csi-provisioner-5bfb9d8577-bq4v8                    1/1     Running   0          72s
> csi-attacher-66d89c889-wmnnl                        1/1     Running   0          72s
> longhorn-csi-plugin-bwhts                           3/3     Running   0          71s
> longhorn-csi-plugin-lh88t                           3/3     Running   0          71s
> longhorn-csi-plugin-ntz9f                           3/3     Running   0          71s
> ```

Following the steps above, Longhorn can success install/uninstall on Ubuntu 22.04 with `v1.21.6+k3s1` by yaml and helm.
